WordPress/src/androidTest/java/org/wordpress/android/e2e/ReaderTests.kt

```diff
@@ -21,30 +21,43 @@ class ReaderTests : BaseTest() {
         ReaderPage().go()
     }
 
-    var mCoachingPostTitle = "Let's check out the coaching team!"
-    var mCompetitionPostTitle = "Let's focus on the competition."
     @Test
     fun e2eNavigateThroughPosts() {
         ReaderPage()
             .tapFollowingTab()
-            .openPost(mCoachingPostTitle)
-            .verifyPostDisplayed(mCoachingPostTitle)
+            .openPost(TITLE_COACHING_POST)
+            .verifyPostDisplayed(TITLE_COACHING_POST)
             .slideToPreviousPost()
-            .verifyPostDisplayed(mCompetitionPostTitle)
+            .verifyPostDisplayed(TITLE_COMPETITION_POST)
             .slideToNextPost()
-            .verifyPostDisplayed(mCoachingPostTitle)
+            .verifyPostDisplayed(TITLE_COACHING_POST)
             .goBackToReader()
     }
 
     @Test
     fun e2eLikePost() {
         ReaderPage()
             .tapFollowingTab()
-            .openPost(mCoachingPostTitle)
+            .openPost(TITLE_COACHING_POST)
             .likePost()
             .verifyPostLiked()
             .unlikePost()
             .verifyPostNotLiked()
             .goBackToReader()
     }
+
+    @Test
+    fun e2eBookmarkPost() {
+        ReaderPage()
+            .openBlog(TITLE_BLOG)
+            .bookmarkPost()
+            .verifyPostBookmarked()
+            .goBackToReader()
+    }
```

Comment:
if we run the test back to back the second time, it will always fail because it's at a state where the post is bookmarked. the state is not cleared when the test ends. 

we could add another step to remove the bookmark before the test ends (or other ways to clear the state, but this seems to be the easiest way and was done for the other test `e2eLikePost`)

```suggestion
            .openBlog(TITLE_BLOG)
            .bookmarkPost()
            .verifyPostBookmarked()
            .unBookmarkPost()
            .goBackToReader()
    }
```

Reply:
95da25e7596912cf8b00613f73062cceed1ff412 I added `removeBookmarkPost ` and  `verifyPostNotBookmarked` to the test steps. Also, I changed the finding views method.